x-pack/plugins/synthetics/server/routes/monitor_cruds/add_monitor_project.ts

```diff
@@ -32,41 +32,46 @@ export const addSyntheticsProjectMonitorRoute: SyntheticsStreamingRouteFactory =
     syntheticsMonitorClient,
     subject,
   }): Promise<any> => {
-    const monitors = (request.body?.monitors as ProjectBrowserMonitor[]) || [];
-    const spaceId = server.spaces.spacesService.getSpaceId(request);
-    const { keep_stale: keepStale, project: projectId } = request.body || {};
-    const { publicLocations, privateLocations } = await getAllLocations(
-      server,
-      syntheticsMonitorClient,
-      savedObjectsClient
-    );
-    const encryptedSavedObjectsClient = server.encryptedSavedObjects.getClient();
+    try {
+      const monitors = (request.body?.monitors as ProjectBrowserMonitor[]) || [];
+      const spaceId = server.spaces.spacesService.getSpaceId(request);
+      const { keep_stale: keepStale, project: projectId } = request.body || {};
+      const { publicLocations, privateLocations } = await getAllLocations(
+        server,
+        syntheticsMonitorClient,
+        savedObjectsClient
+      );
+      const encryptedSavedObjectsClient = server.encryptedSavedObjects.getClient();
 
-    const pushMonitorFormatter = new ProjectMonitorFormatter({
-      projectId,
-      spaceId,
-      keepStale,
-      locations: publicLocations,
-      privateLocations,
-      encryptedSavedObjectsClient,
-      savedObjectsClient,
-      monitors,
-      server,
-      syntheticsMonitorClient,
-      request,
-      subject,
-    });
+      const pushMonitorFormatter = new ProjectMonitorFormatter({
+        projectId,
+        spaceId,
+        keepStale,
+        locations: publicLocations,
+        privateLocations,
+        encryptedSavedObjectsClient,
+        savedObjectsClient,
+        monitors,
+        server,
+        syntheticsMonitorClient,
+        request,
+        subject,
+      });
 
-    await pushMonitorFormatter.configureAllProjectMonitors();
+      await pushMonitorFormatter.configureAllProjectMonitors();
 
-    subject?.next({
-      createdMonitors: pushMonitorFormatter.createdMonitors,
-      updatedMonitors: pushMonitorFormatter.updatedMonitors,
-      staleMonitors: pushMonitorFormatter.staleMonitors,
-      deletedMonitors: pushMonitorFormatter.deletedMonitors,
-      failedMonitors: pushMonitorFormatter.failedMonitors,
-      failedStaleMonitors: pushMonitorFormatter.failedStaleMonitors,
-    });
-    subject?.complete();
+      subject?.next({
+        createdMonitors: pushMonitorFormatter.createdMonitors,
+        updatedMonitors: pushMonitorFormatter.updatedMonitors,
+        staleMonitors: pushMonitorFormatter.staleMonitors,
+        deletedMonitors: pushMonitorFormatter.deletedMonitors,
+        failedMonitors: pushMonitorFormatter.failedMonitors,
+        failedStaleMonitors: pushMonitorFormatter.failedStaleMonitors,
+      });
+    } catch (error) {
+      subject?.error(error);
+    } finally {
```

Comment:
I thought I added this? I guess I didn't push my last commit. :(

Reply:
No worries, we got it in there 😄